src/omitBy.test.ts

```diff
@@ -0,0 +1,26 @@
+import { omitBy } from './omitBy';
+import { pipe } from './pipe';
+
+describe('data first', () => {
+  test('it should pick props', () => {
+    const result = omitBy(
+      { a: 1, b: 2, A: 3, B: 4 },
+      (val, key) => key.toUpperCase() === key
+    );
+    expect(result).toStrictEqual({ a: 1, b: 2 });
+  });
+  test('allow undefined or null', () => {
+    expect(omitBy(undefined as any, (val, key) => key === 'foo')).toEqual({});
+    expect(omitBy(null as any, (val, key) => key === 'foo')).toEqual({});
+  });
```

Comment:
```suggestion
  });
  test('allow partial type', () => {
    expect(omitBy({} as Partial<{ a: string, b: number }>, (val, key) => key === 'a')).toEqual({});
  });
```

Reply:
That wouldn't work. OmitBy like a PickBy works only with Record

Reply:
I think both OmitBy/PickBy should also work with arbitrary object kind, if possible. It is sometimes impossible to control input argument.

One hackish way to use a typed object with rameda will looks like this :--

```ts
const data = {} as Partial<{ a: string }>;
const result = R.omitBy(
  (data as unknown) as Record<keyof typeof data, unknown>,
  (v, k) => k === "a"
);
```

Reply:
I understood your problem. Try to implement workaround & cover with tests. I'll let you know when finish

Reply:
@sushantdhiman Done :)

Reply:
@vlad-yakovlev Thanks 👍🏻